### PR TITLE
Used Zustand to implement State Management

### PR DESCRIPTION
**Pull request by denimozh**
Used Zustand to implement State Management between the cart component

### CODE REVIEW

File: src/app/api/catalog/route.ts [hunk 1]
```diff
@@ -1,20 +1,21 @@
...
         const catalog = await prisma.category.findMany();
 
-        return new NextResponse(JSON.stringify(catalog), {status:200});
-
+        return new NextResponse(JSON.stringify(catalog), { status: 200 });
     } catch (error) {
         console.log(error);
-        return new NextResponse(JSON.stringify({message:"Something went wrong!"}), {status:500});
+        return new NextResponse(
+            JSON.stringify({ message: 'Something went wrong!' }),
```
Comment (AliakseiDeboi) [inline src/app/api/catalog/route.ts:13]: Enum for constant texts/values

File: src/app/orders/page.tsx [hunk 2]
```diff
@@ -1,9 +1,122 @@
...
+        queryFn: async () =>
+            await fetch('http://localhost:3000/api/orders').then(
+                async res => await res.json(),
+            ),
+    });
+
+    const handleUpdate = (e: React.FormEvent<HTMLFormElement>, id: string) => {
+        e.preventDefault();
+        const form = e.target as HTMLFormElement;
+        const input = form.elements[0] as HTMLInputElement;
```
Comment (AliakseiDeboi) [inline src/app/orders/page.tsx:48]: what if form is not exists?

**Review by AliakseiDeboi — APPROVED** [review]

File: src/app/api/catalog/route.ts (hunk 1 above)
Comment (denimozh) [reply src/app/api/catalog/route.ts:13]: Ah yes I will make Enums next

File: src/app/orders/page.tsx (hunk 2 above)
Comment (denimozh) [reply src/app/orders/page.tsx:48]: Form will always exist as it is part of the 'component' - the buttons will always be there

**Pull request merged**